src/App.tsx

```diff
@@ -1,34 +1,101 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Completed, Filters } from './types/Filters';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | undefined>(undefined);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [filters, setFilters] = useState<Filters>({
+    completedType: Completed.All,
+    searchByText: '',
+  });
+
+  const selectTodo = (todo: Todo | null) => {
+    setSelectedTodo(todo);
+  };
+
+  const updateFilters = (key: keyof Filters, value: string | Completed) => {
+    setFilters(prevFilters => ({
+      ...prevFilters,
+      [key]: value,
+    }));
+  };
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(item => {
+      const matchesCompletedType =
+        filters.completedType === Completed.All ||
+        (filters.completedType === Completed.Completed && item.completed) ||
+        (filters.completedType === Completed.Active && !item.completed);
+
+      const matchesSearchText = item.title
+        .toLowerCase()
+        .includes(filters.searchByText.toLowerCase());
+
+      return matchesCompletedType && matchesSearchText;
+    });
+  }, [todos, filters]);
+
+  useEffect(() => {
+    getTodos()
+      .then(todosData => {
+        setTodos(todosData);
+      })
+      .catch(error => {
+        setErrorMessage(`Failed to fetch todos: ${error.message}`);
+      });
+  }, []);
+
   return (
     <>
       <div className="section">
         <div className="container">
           <div className="box">
             <h1 className="title">Todos:</h1>
 
+            {errorMessage && (
+              <p className="notification is-danger">{errorMessage}</p>
+            )}
+
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filterOptions={filters}
+                onFilterChange={updateFilters}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0 ? (
+                <Loader />
```

Comment:
The `Loader` component is displayed when `todos.length === 0`. This logic might cause the loader to appear even when there are no todos to fetch or display. Consider checking if the data is still being fetched to conditionally render the `Loader` component.